src/lib/components/data-graphic/compositions/timestamp-detail/TimestampSpark.svelte

```diff
@@ -0,0 +1,175 @@
+<script lang="ts">
+  /**
+   * TimestampSpark.svelte
+   * ---------------------
+   * This simple component is a basic sparkline, meant to be used
+   * in a table / model profile preview.
+   * It optionally enables the user to determine a "window", which
+   * is just a box emcompassing the zoomWindowXMin and zoomWindowXMax values.
+   */
+  import { guidGenerator } from "$lib/util/guid";
+  import { fade } from "svelte/transition";
+  import { cubicOut as easing } from "svelte/easing";
+  import { scaleLinear } from "d3-scale";
+  import { extent } from "d3-array";
+  import { writable } from "svelte/store";
+  import { createExtremumResolutionStore } from "../../extremum-resolution-store";
+  import { lineFactory, areaFactory } from "./utils";
+  import { tweened } from "svelte/motion";
+
+  const plotID = guidGenerator();
+
+  export let data;
+
+  export let width = 360;
+  export let height = 120;
+  export let curve = "curveLinear";
+  export let area = false;
+  export let color = "hsl(217, 10%, 50%)";
+  export let tweenIn = false;
+
+  // the color of the zoom window
+  export let zoomWindowColor = "hsla(217, 90%, 60%, .2)";
+  // the color of the zoom window boundaries
+  export let zoomWindowBoundaryColor = "rgb(100,100,100)";
+  export let zoomWindowXMin: any = undefined;
+  export let zoomWindowXMax: any = undefined;
+
+  export let xAccessor: string = undefined;
+  export let yAccessor: string = undefined;
+
+  // rowsize for table
+  export let left = 4;
+  export let right = 4;
+  export let top = 12;
+  export let bottom = 4;
+
+  export let buffer = 4;
+
+  const X = writable(undefined);
+  const Y = writable(undefined);
+
+  $: plotTop = top + buffer;
+  $: plotBottom = height - buffer - bottom;
+  $: plotLeft = left + buffer;
+  $: plotRight = width - right - buffer;
+
+  // establish basis values
+  let xExtents = extent(data, (d) => d[xAccessor]);
+  $: xExtents = extent(data, (d) => d[xAccessor]);
+
+  const xMin = createExtremumResolutionStore(xExtents[0], {
+    duration: 300,
+    easing,
+    direction: "min",
+  });
+  const xMax = createExtremumResolutionStore(xExtents[1], {
+    duration: 300,
+    easing,
+  });
+
+  $: xMin.setWithKey("x", xExtents[0]);
+  $: xMax.setWithKey("x", xExtents[1]);
+
+  // Let's set the X Scale based on the $xMin and $xMax.
+  $: $X = scaleLinear()
```

Comment:
yep that's correct

Reply:
happy to change this to `X.set(...)` instead since it's a bit easier to follow along

Reply:
Nah, it's just part of my learning curve, just had to ask to make sure since I couldn't find a reference for that. If this syntax is the best practice for Svelte, I think it's fine to use it -- whatever is most readable to people who know Svelte, and the rest of us will learn + help new folks learn :-)

Reply:
I think both approaches are usually taken. I had originally defined `X` as a normal reactive variable, then switched it to a store for the sake of future portability. I almost always prefer to use `X.set(...)` where I can to be as unconfusing as possible.